addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2237,13 +2238,32 @@ export class OdooEditor extends EventTarget {
                 // Check for url after user insert a space so we won't transform an incomplete url.
                 if (
                     ev.data &&
-                    ev.data.includes(' ') &&
+                    ev.data === ' ' &&
                     selection &&
-                    selection.anchorNode
-                    && (!this.commandBar._active ||
+                    selection.anchorNode &&
+                    !closestElement(selection.anchorNode).closest('a') &&
+                    selection.anchorNode.nodeType === Node.TEXT_NODE &&
+                    (!this.commandBar._active ||
                         this.commandBar._currentOpenOptions.closeOnSpace !== true)
                 ) {
-                    this._convertUrlInElement(closestElement(selection.anchorNode));
+                    const text = selection.anchorNode.textContent.slice(0, selection.anchorOffset);
+                    const textNodeSplitted = text.split(/\s/);
+
+                    // Remove added space
+                    textNodeSplitted.pop();
+                    const potentialUrl = textNodeSplitted.pop();
+                    const lastWordMatch = new RegExp(URL_REGEX_WITH_INFOS).exec(potentialUrl);
+
+                    if (lastWordMatch) {
+                        const matches = getUrlsInfosInString(selection.anchorNode.textContent.slice(0, selection.anchorOffset));
```

Comment:
It is not clear to me why you need to re-slice considering you already did it earlier.